src/praisonai/praisonai/bots/_defaults.py

```
@@ -0,0 +1,209 @@
+            try:
+                resolved_tools = _resolve_tool_names(default_safe_tools)
+                if resolved_tools:
+                    agent.tools = resolved_tools
+                    logger.debug(f"Bot: applied {len(resolved_tools)} default safe tools to agent '{getattr(agent, 'name', '?')}'")
+                else:
+                    # Profile resolution returned empty - try fallback
+                    fallback_tools = _get_fallback_tools()
+                    if fallback_tools:
+                        agent.tools = fallback_tools
+                        logger.debug(f"Bot: applied {len(fallback_tools)} fallback tools to agent '{getattr(agent, 'name', '?')}'")
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)